2021-08-26 13:36:09.439 [http-nio-8080-exec-9] ERROR onlinepharmacy.controller.command.impl.RegistrationCommand - Exception is in RegistrationCommand 
by.epam.onlinepharmacy.exception.ServiceException: Exception is in method createUser(), when send code 
	at by.epam.onlinepharmacy.model.service.impl.UserServiceImpl.createUser(UserServiceImpl.java:74) ~[classes/:?]
	at by.epam.onlinepharmacy.controller.command.impl.RegistrationCommand.execute(RegistrationCommand.java:33) [classes/:?]
	at by.epam.onlinepharmacy.controller.Controller.processRequest(Controller.java:33) [classes/:?]
	at by.epam.onlinepharmacy.controller.Controller.doPost(Controller.java:25) [classes/:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:652) [servlet-api.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:733) [servlet-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) [catalina.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.57]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-websocket.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) [catalina.jar:8.5.57]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139) [catalina.jar:8.5.57]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) [catalina.jar:8.5.57]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:690) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87) [catalina.jar:8.5.57]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343) [catalina.jar:8.5.57]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:615) [tomcat-coyote.jar:8.5.57]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:8.5.57]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:818) [tomcat-coyote.jar:8.5.57]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1626) [tomcat-coyote.jar:8.5.57]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:8.5.57]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.5.57]
	at java.lang.Thread.run(Thread.java:831) [?:?]
Caused by: by.epam.onlinepharmacy.exception.DaoException: SQLException in method createUser() 
	at by.epam.onlinepharmacy.model.dao.impl.UserDaoImpl.findByLogin(UserDaoImpl.java:107) ~[classes/:?]
	at by.epam.onlinepharmacy.model.service.impl.UserServiceImpl.createUser(UserServiceImpl.java:66) ~[classes/:?]
	... 27 more
Caused by: java.sql.SQLException: Column 'email' not found.
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.result.ResultSetImpl.findColumn(ResultSetImpl.java:556) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:863) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at by.epam.onlinepharmacy.model.dao.impl.UserDaoImpl.findByLogin(UserDaoImpl.java:100) ~[classes/:?]
	at by.epam.onlinepharmacy.model.service.impl.UserServiceImpl.createUser(UserServiceImpl.java:66) ~[classes/:?]
	... 27 more
2021-08-26 13:38:08.154 [http-nio-8080-exec-6] ERROR onlinepharmacy.model.dao.impl.UserDaoImpl - SQLException in method createUser() 
java.sql.SQLException: Column 'email' not found.
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.result.ResultSetImpl.findColumn(ResultSetImpl.java:556) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:863) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at by.epam.onlinepharmacy.model.dao.impl.UserDaoImpl.findByLogin(UserDaoImpl.java:100) [classes/:?]
	at by.epam.onlinepharmacy.model.service.impl.UserServiceImpl.createUser(UserServiceImpl.java:66) [classes/:?]
	at by.epam.onlinepharmacy.controller.command.impl.RegistrationCommand.execute(RegistrationCommand.java:33) [classes/:?]
	at by.epam.onlinepharmacy.controller.Controller.processRequest(Controller.java:33) [classes/:?]
	at by.epam.onlinepharmacy.controller.Controller.doPost(Controller.java:25) [classes/:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:652) [servlet-api.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:733) [servlet-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) [catalina.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.57]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-websocket.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) [catalina.jar:8.5.57]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139) [catalina.jar:8.5.57]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) [catalina.jar:8.5.57]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:690) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87) [catalina.jar:8.5.57]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343) [catalina.jar:8.5.57]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:615) [tomcat-coyote.jar:8.5.57]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:8.5.57]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:818) [tomcat-coyote.jar:8.5.57]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1626) [tomcat-coyote.jar:8.5.57]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:8.5.57]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.5.57]
	at java.lang.Thread.run(Thread.java:831) [?:?]
2021-08-26 13:38:08.156 [http-nio-8080-exec-6] ERROR onlinepharmacy.model.service.impl.UserServiceImpl - Exception is in method createUser(), when send code
by.epam.onlinepharmacy.exception.DaoException: SQLException in method createUser() 
	at by.epam.onlinepharmacy.model.dao.impl.UserDaoImpl.findByLogin(UserDaoImpl.java:107) ~[classes/:?]
	at by.epam.onlinepharmacy.model.service.impl.UserServiceImpl.createUser(UserServiceImpl.java:66) [classes/:?]
	at by.epam.onlinepharmacy.controller.command.impl.RegistrationCommand.execute(RegistrationCommand.java:33) [classes/:?]
	at by.epam.onlinepharmacy.controller.Controller.processRequest(Controller.java:33) [classes/:?]
	at by.epam.onlinepharmacy.controller.Controller.doPost(Controller.java:25) [classes/:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:652) [servlet-api.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:733) [servlet-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) [catalina.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.57]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-websocket.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) [catalina.jar:8.5.57]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139) [catalina.jar:8.5.57]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) [catalina.jar:8.5.57]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:690) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87) [catalina.jar:8.5.57]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343) [catalina.jar:8.5.57]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:615) [tomcat-coyote.jar:8.5.57]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:8.5.57]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:818) [tomcat-coyote.jar:8.5.57]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1626) [tomcat-coyote.jar:8.5.57]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:8.5.57]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.5.57]
	at java.lang.Thread.run(Thread.java:831) [?:?]
Caused by: java.sql.SQLException: Column 'email' not found.
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.result.ResultSetImpl.findColumn(ResultSetImpl.java:556) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:863) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at by.epam.onlinepharmacy.model.dao.impl.UserDaoImpl.findByLogin(UserDaoImpl.java:100) ~[classes/:?]
	... 28 more
2021-08-26 13:38:08.163 [http-nio-8080-exec-6] ERROR onlinepharmacy.controller.command.impl.RegistrationCommand - Exception is in RegistrationCommand 
by.epam.onlinepharmacy.exception.ServiceException: Exception is in method createUser(), when send code 
	at by.epam.onlinepharmacy.model.service.impl.UserServiceImpl.createUser(UserServiceImpl.java:74) ~[classes/:?]
	at by.epam.onlinepharmacy.controller.command.impl.RegistrationCommand.execute(RegistrationCommand.java:33) [classes/:?]
	at by.epam.onlinepharmacy.controller.Controller.processRequest(Controller.java:33) [classes/:?]
	at by.epam.onlinepharmacy.controller.Controller.doPost(Controller.java:25) [classes/:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:652) [servlet-api.jar:?]
	at javax.servlet.http.HttpServlet.service(HttpServlet.java:733) [servlet-api.jar:?]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:231) [catalina.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.57]
	at org.apache.tomcat.websocket.server.WsFilter.doFilter(WsFilter.java:52) [tomcat-websocket.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.internalDoFilter(ApplicationFilterChain.java:193) [catalina.jar:8.5.57]
	at org.apache.catalina.core.ApplicationFilterChain.doFilter(ApplicationFilterChain.java:166) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardWrapperValve.invoke(StandardWrapperValve.java:199) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardContextValve.invoke(StandardContextValve.java:96) [catalina.jar:8.5.57]
	at org.apache.catalina.authenticator.AuthenticatorBase.invoke(AuthenticatorBase.java:543) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardHostValve.invoke(StandardHostValve.java:139) [catalina.jar:8.5.57]
	at org.apache.catalina.valves.ErrorReportValve.invoke(ErrorReportValve.java:81) [catalina.jar:8.5.57]
	at org.apache.catalina.valves.AbstractAccessLogValve.invoke(AbstractAccessLogValve.java:690) [catalina.jar:8.5.57]
	at org.apache.catalina.core.StandardEngineValve.invoke(StandardEngineValve.java:87) [catalina.jar:8.5.57]
	at org.apache.catalina.connector.CoyoteAdapter.service(CoyoteAdapter.java:343) [catalina.jar:8.5.57]
	at org.apache.coyote.http11.Http11Processor.service(Http11Processor.java:615) [tomcat-coyote.jar:8.5.57]
	at org.apache.coyote.AbstractProcessorLight.process(AbstractProcessorLight.java:65) [tomcat-coyote.jar:8.5.57]
	at org.apache.coyote.AbstractProtocol$ConnectionHandler.process(AbstractProtocol.java:818) [tomcat-coyote.jar:8.5.57]
	at org.apache.tomcat.util.net.NioEndpoint$SocketProcessor.doRun(NioEndpoint.java:1626) [tomcat-coyote.jar:8.5.57]
	at org.apache.tomcat.util.net.SocketProcessorBase.run(SocketProcessorBase.java:49) [tomcat-coyote.jar:8.5.57]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1130) [?:?]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:630) [?:?]
	at org.apache.tomcat.util.threads.TaskThread$WrappingRunnable.run(TaskThread.java:61) [tomcat-util.jar:8.5.57]
	at java.lang.Thread.run(Thread.java:831) [?:?]
Caused by: by.epam.onlinepharmacy.exception.DaoException: SQLException in method createUser() 
	at by.epam.onlinepharmacy.model.dao.impl.UserDaoImpl.findByLogin(UserDaoImpl.java:107) ~[classes/:?]
	at by.epam.onlinepharmacy.model.service.impl.UserServiceImpl.createUser(UserServiceImpl.java:66) ~[classes/:?]
	... 27 more
Caused by: java.sql.SQLException: Column 'email' not found.
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:129) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:97) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:89) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:63) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.result.ResultSetImpl.findColumn(ResultSetImpl.java:556) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at com.mysql.cj.jdbc.result.ResultSetImpl.getString(ResultSetImpl.java:863) ~[mysql-connector-java-8.0.20.jar:8.0.20]
	at by.epam.onlinepharmacy.model.dao.impl.UserDaoImpl.findByLogin(UserDaoImpl.java:100) ~[classes/:?]
	at by.epam.onlinepharmacy.model.service.impl.UserServiceImpl.createUser(UserServiceImpl.java:66) ~[classes/:?]
	... 27 more
